dev/com.ibm.ws.transport.http/src/com/ibm/ws/http/logging/source/AccessLogSource.java

```diff
@@ -263,180 +309,188 @@ AccessLogDataFormatter populateCustomFormatters(FormatSegment[] parsedFormat, Fo
-            if (currentSF.checkConfigChange(formatString, jsonAccessLogFields, jsonAccessLogFieldsLogstash)) {
-                SetterFormatter newSF = new SetterFormatter();
-                List<AccessLogDataFieldSetter> fieldSetters = new ArrayList<AccessLogDataFieldSetter>();
-                AccessLogDataFormatter[] formatters = { null, null, null, null };
-                Map<String, HashSet<Object>> fieldsToAdd = new HashMap<String, HashSet<Object>>();
-
-                // Create the mapping of fields to add:{<format key> : <data value/null>}
-                // Prevents duplicates
-                initializeFieldMap(fieldsToAdd, parsedFormat);
-
-                newSF.setFullConfig(formatString, jsonAccessLogFields, jsonAccessLogFieldsLogstash);
-
-                // Create setter list
-                fieldSetters = populateSetters(fieldsToAdd);
-                // These fields are always added
-                fieldSetters.add((ald, alrd) -> ald.setSequence(alrd.getStartTime() + "_" + String.format("%013X", seq.incrementAndGet())));
-                fieldSetters.add((ald, alrd) -> ald.setDatetime(alrd.getTimestamp()));
-
-                if (jsonAccessLogFields.equals("default")) {
-                    formatters[0] = populateDefaultFormatters(FormatSpecifier.JSON);
-                }
-                if (jsonAccessLogFields.equals("logFormat")) {
-                    formatters[1] = populateCustomFormatters(parsedFormat, FormatSpecifier.JSON);
-                }
-                if (jsonAccessLogFieldsLogstash.equals("default")) {
-                    formatters[2] = populateDefaultFormatters(FormatSpecifier.LOGSTASH);
-                }
-                if (jsonAccessLogFieldsLogstash.equals("logFormat")) {
-                    formatters[3] = populateCustomFormatters(parsedFormat, FormatSpecifier.LOGSTASH);
+        public void process(AccessLogRecordData recordData) {
+            // The logFormat, as a string: e.g. "%a %b %C"
+            String formatString = ((AccessLogRecordDataExt) recordData).getFormatString();
+            // A parsed version of the logFormat
+            FormatSegment[] parsedFormat = ((AccessLogRecordDataExt) recordData).getParsedFormat();
+            jsonAccessLogFieldsConfig = AccessLogConfig.jsonAccessLogFieldsConfig;
+
+            Configuration config = new Configuration(formatString, jsonAccessLogFieldsConfig, jsonAccessLogFieldsLogstashConfig);
+
+            if (currentSF.checkConfigChange(formatString, jsonAccessLogFieldsConfig, jsonAccessLogFieldsLogstashConfig))
+                if (setterFormatterMap.containsKey(config)) {
+                    // If we've created a setterFormatter in the past for this configuration, we'll use it instead of making a new one
+                    currentSF = setterFormatterMap.get(config);
+                } else {
+                    initializeSetterFormatter(formatString, jsonAccessLogFieldsConfig, jsonAccessLogFieldsLogstashConfig, parsedFormat, seq);
```

Comment:
change this method to be static and just return the SetterFormatter it creates.  rename to createSetterFormatter